### PR TITLE
3.3.0 remove a layer of indirection (unnecessary passthrough)

### PR DESCRIPTION
**Pull request by jackyzha0**
we shouldnt use passthrough if we dont need it, we got wrecked by the default watermark on the passthrough stream

### CODE REVIEW

File: wrapper.ts [hunk 1]
```diff
@@ -70,27 +73,27 @@ export class Pty {
...
 
-    this.#socket = new ReadStream(this.#fd);
-    const userFacingRead = new PassThrough();
-    const userFacingWrite = new PassThrough();
-    this.#socket.pipe(userFacingRead);
-    userFacingWrite.pipe(this.#socket);
-    this.read = userFacingRead;
-    this.write = userFacingWrite;
+    this.read = new ReadStream(this.#fd);
+    this.write = new WriteStream(this.#fd);
```
Comment (lhchavez) [inline wrapper.ts:77]: warning: the pty implementation handles muxing both internally. please make sure this is safe and won't lead to the FD being closed twice!

**Review by blast-hardcheese — APPROVED** [review]

File: wrapper.ts [hunk 2]
```diff
@@ -70,27 +79,29 @@ export class Pty {
...
-    const userFacingRead = new PassThrough();
-    const userFacingWrite = new PassThrough();
-    this.#socket.pipe(userFacingRead);
-    userFacingWrite.pipe(this.#socket);
-    this.read = userFacingRead;
-    this.write = userFacingWrite;
+    this.#socket = new ReadStream(this.#fd)
+    this.write = new Writable({
+      write: this.#socket.write.bind(this.#socket),
+    });
```
Comment (lhchavez) [inline wrapper.ts:85]: this looks disgusting. i love it.

File: wrapper.ts [hunk 3]
```diff
@@ -45,21 +45,30 @@ type ExitResult = {
...
+  #handledClose: boolean = false;
+  #handledEndOfData: boolean = false;
+
   #socket: ReadStream;
+  get read(): Readable {
+    return this.#socket;
+  }
 
-  read: Readable;
   write: Writable;
```
Comment (lhchavez) [inline wrapper.ts:57]: can we also have this be a getter for symmetry?

File: wrapper.ts [hunk 4]
```diff
@@ -70,27 +79,29 @@ export class Pty {
...
     this.#fd = this.#pty.takeFd();
 
-    this.#socket = new ReadStream(this.#fd);
-    const userFacingRead = new PassThrough();
-    const userFacingWrite = new PassThrough();
-    this.#socket.pipe(userFacingRead);
-    userFacingWrite.pipe(this.#socket);
-    this.read = userFacingRead;
-    this.write = userFacingWrite;
+    this.#socket = new ReadStream(this.#fd)
```
Comment (lhchavez) [inline wrapper.ts:82]:
```suggestion
    this.#socket = new ReadStream(this.#fd);
```

ewwww semicolonless code.

**Review by lhchavez — APPROVED** [review]

**Pull request merged**